src/field/vecops.cxx

```diff
@@ -34,69 +34,85 @@
  * Gradient operators
  **************************************************************************/
 
-const Vector2D Grad(const Field2D &f, CELL_LOC UNUSED(outloc)) {
+const Vector2D Grad(const Field2D &f, CELL_LOC outloc) {
   Vector2D result(f.getMesh());
 
   TRACE("Grad( Field2D )");
   
-  result.x = DDX(f);
-  result.y = DDY(f);
-  result.z = DDZ(f);
+  if (outloc == CELL_DEFAULT) {
```

Comment:
Can we just pass `CELL_DEFAULT` through to the derivative operators? Ideally the interpretation of `CELL_DEFAULT` can be minimised to just the lowest level where it's actually needed.

Reply:
This is ok now. `CELL_DEFAULT` is passed down to `DD?`, and is then used to set the location of the result to the location of `result.x`